go/example/om-agones-game-frontend/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"open-match.dev/open-match/pkg/pb"
+	"google.golang.org/grpc"
+	"context"
+	"time"
+	"fmt"
+
+)
+
+func createOMFrontendClient() (pb.FrontendClient, func() error) {
+	conn, err := grpc.Dial("om-frontend.open-match.svc.cluster.local:50504", grpc.WithInsecure())
+	if err != nil {
+		panic(err)
+	}
+	return pb.NewFrontendClient(conn), conn.Close
+}
+
+func createOMTicket() *pb.Ticket {
+	return &pb.Ticket{
+		SearchFields: &pb.SearchFields{
+			DoubleArgs: map[string]float64{"defense": 10, "attack": 20, "level": 50},
+			StringArgs: map[string]string{"guildName": "open-match", "location": "alterac"},
+			Tags: []string{"beta-gameplay", "demo-map"},
+		},
+	}
+}
+
+func main() {
+	fc, closer := createOMFrontendClient()
+	defer closer()
+
+	resp, err := fc.CreateTicket(context.Background(), &pb.CreateTicketRequest{Ticket: createOMTicket()})
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// Each channel represents a player waiting for a game, the channel will be closed if the player exit the queue
+	c := make(chan bool, 1)
+	tid := resp.GetTicket().GetId()
+	for {
+		// TODO: once we find a game, have the game send something to this channel once the player starts a match making
+		_, ok := <- c
```

Comment:
This will block forever: https://play.golang.org/p/ovKSxjmoSX2

Reply:
ideally somewhere in the game we should have something send to this channel to unblock it. This is just a baseline of the game frontend - the final shape of it depends on the actual game that we use.